### PR TITLE
Refactor spinnaker to VMSS template

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
1. Move to one VM to reduce the cost by half
2. Use the azure-devops-utils repo for scripts
3. Setup CI for this template
4. Reduce the number of parameters needed
5. Move to the up-to-date Jenkins template instead of using the out-of-date jenkins image
6. Automate a lot of steps, including:
  a. Install Java JDK without using oracle username/password
  b. Setup aptly and serve it on 9999
  c. Setup first Jenkins account
  d. Add a sample Jenkins job that publishs to Aptly

### CODE REVIEW

File: 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    "subnet0Name": "[concat(variables('scenarioPrefix'), 'Subnet0')]",
+    "subnet1Name": "[concat(variables('scenarioPrefix'), 'Subnet1')]",
+    "subnet2Name": "[concat(variables('scenarioPrefix'), 'Subnet2')]",
+    "frontEndNSGName": "[concat(variables('scenarioPrefix'), 'NSG')]",
+    "nicName": "[concat(variables('scenarioPrefix'), 'Nic')]",
+    "publicIPAddressName": "[concat(variables('scenarioPrefix'), 'PublicIP')]",
+    "vmName": "[concat(variables('scenarioPrefix'), 'VM')]",
+    "vmExtensionName": "[concat(variables('scenarioPrefix'), 'Init')]",
+    "spinnakerOSDiskName": "[concat(variables('scenarioPrefix'), 'OSDisk')]",
+    "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-devops-utils/vmss/",
```
Comment (EricJizbaMSFT) [inline 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json]: Note to self: Update this to the released version of utils repo

File: 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+        "description": "Service Principal App ID (also called Client ID) that has contributor rights to the subscription used for this deployment. It is used by Spinnaker to dynamically manage resources."
+      }
+    },
+    "servicePrincipalAppKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Service Principal App Key (also called Client Secret) that has contributor rights to the subscription used for this deployment. It is used by Spinnaker to dynamically manage resources."
+      }
+    },
+    "jenkinsUsername": {
```
Comment (clguimanMSFT) [inline 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json]: just ignore: I don't like that there are 2 types of users, but I do understand why :)

Comment (EricJizbaMSFT) [inline 301-jenkins-aptly-spinnaker-vmss/azuredeploy.json]: I don't like it either. This template is much more of a 'try it out' scenario than a 'production ready' scenario, so I think it's fine to remove these and just use the admin password.

**Comment by dcaro** [comment]
LGTM.

**Pull request merged**